test/util/util.go

```diff
@@ -54,61 +53,31 @@ import (
 	"sigs.k8s.io/kueue/pkg/workload"
 )
 
-func DeleteAdmissionCheck(ctx context.Context, c client.Client, ac *kueue.AdmissionCheck) error {
-	if ac != nil {
-		if err := c.Delete(ctx, ac); err != nil && !apierrors.IsNotFound(err) {
-			return err
-		}
-	}
-	return nil
-}
-
-func DeleteProvisioningRequestConfig(ctx context.Context, c client.Client, ac *kueue.ProvisioningRequestConfig) error {
-	if ac != nil {
-		if err := c.Delete(ctx, ac); err != nil && !apierrors.IsNotFound(err) {
-			return err
-		}
-	}
-	return nil
+type objAsPtr[T any] interface {
+	client.Object
+	*T
 }
 
-func DeleteWorkload(ctx context.Context, c client.Client, wl *kueue.Workload) error {
-	if wl != nil {
-		if err := c.Delete(ctx, wl); err != nil && !apierrors.IsNotFound(err) {
+func DeleteObject[PtrT objAsPtr[T], T any](ctx context.Context, c client.Client, o PtrT) error {
```

Comment:
```suggestion
func DeleteObject(ctx context.Context, c client.Client, o client.Object) error {
```

Reply:
We'll louse the possibility to do the nil check (on the concrete object), case in which the delete will fail.  
Also the function will just be an alias to "c.Delete(ctx, o)"

Reply:
Not really, you can detect if it's called `Delete(..., nil)` but not for `Delete(..., (*kueue.Workload)(nil))`